server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -374,6 +382,13 @@ public Query getFieldQuery(String field, String queryText, boolean quoted) throw
         }
     }
 
+    private void addDiscoveredField(String field) {
```

Comment:
if we have to keep invoke `addDiscoveredField` in every method manually we can't guarantee that this wont be missed in any methods added in the future?

Reply:
That could be an issue.

The only alternative I can think of is maybe traversing resulting Query "tree", but there also we could add new types of queries and miss some fields..